### ISSUE TITLE
Allow setting x-forwarded-host and x-forwarded-port

### ISSUE THREAD

**Issue by totallyGreg** [solo-io/gloo#7683]
### Version

1.12.x

### Is your feature request related to a problem? Please describe.

Customer requires exposing `x-forwarded-host` and `x-forwarded-port` to their upstream.  

Although Envoy has this ability, it is not currently exposed via the Gloo Edge API.

### Describe the solution you'd like

A simple way to do this via the Gloo Edge API

### Describe alternatives you've considered

As discussed here: https://solo-io-corp.slack.com/archives/C02LQ0JCNLF/p1638181079038900?thread_ts=1637688318.011000&cid=C02LQ0JCNLF

It may be possible to define via allowedHeaders e.g.
```
global:
  extensions:
    extAuth:
      enabled: false
gloo:
  settings:
    kubeResourceOverride:
      spec:
        extauth:
          extauthzServerRef:
            name: external-auth-server
            namespace: kube-gloo-system
          httpService:
            pathPrefix: /envoy/verify-params-header
            request:
              allowedHeaders:
              - cookie
              - x-forwarded-host
              - x-forwarded-method
              - x-forwarded-proto
              - x-forwarded-uri
              - x-ostenan
```

**Comment by SuperAayush** [comment]
Hey I would like to work on this 😄!!

**Comment by SantoDE** [comment]
To be more precise, this is not about explicitly setting those but rather an option to enable that envoy sets them automatically. Similar to what's done in Traefik (https://github.com/solo-io/gloo-mesh-enterprise/issues/9894). 

If it's enabled, envoy could automatically add those headers. See how envoy can do it [click](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/headers#x-forwarded-for)

**Comment by nfuden** [comment]
in 1.15.0-beta4

**Comment by elcasteel** [comment]
This is in gloo oss 1.15.18 and gloo-ee 1.15.4

**Comment by SantoDE** [comment]
Reopening since it needs a back port

**Comment by sam-heilbron** [comment]
The linked PR's have merged the code into 1.15. We need to backport this to the other release versions listed in the labels (release/1.14, release/1.13)

**Comment by nfuden** [comment]
We are not backporting as is to 1.13 and will spin up a separate issue for a 1.13 fork if truly needed.
This functionality relies on a newer version of envoy than 1.13 has so it would imply a lts envoy bump OR a solo owned patch of envoy to enable this behavior. Either is a bigger ask and something that we need to think about critically as any drift raises risk on LTS branches

**Comment by inFocus7** [comment]
Actually, will be re-opening to handle backporting to 1.14 EE

**Comment by inFocus7** [comment]
Released in OSS [v1.14.18](https://github.com/solo-io/gloo/releases/tag/v1.14.18). To-be-released in EE v1.14.9.

**Comment by spunuru** [comment]
In both Gloo Edge 1.15.7 and 1.15.12, I still do not see `x-forwarded-host `and `x-forwarded-port` request headers in my pod. Is this behind a feature flag or do I need to turn on something? The Spring based Java applications that I use need these headers to build URLs to render in response or redirect the users.
